source/Jobbr.Server.UnitTests/Core/TriggerServiceTests.cs

```diff
@@ -0,0 +1,214 @@
+        }
+
+        [TestMethod]
+        public void Add_RecurringTrigger_ShouldCallMessengerHubPublish()
+        {
+            // Arrange
+            var model = new RecurringTriggerModel { Id = 1, JobId = 2 };
+
+            // Act
+            _service.Add(1L, model);
+
+            // Assert
+            _messengerHubMock.Verify(hub => hub.PublishAsync(It.Is<TriggerAddedMessage>(message => message.TriggerId == 1 && message.JobId == 2)), Times.Once());
+        }
+
+        [TestMethod]
+        public void Add_ScheduledTrigger_ShouldSetModelKeys()
+        {
+            // Arrange
+            var triggerModel = new ScheduledTriggerModel { Id = 1, JobId = 2 };
+            _repositoryMock
+                .Setup(rep => rep.SaveAddTrigger(It.IsAny<long>(), It.IsAny<ScheduledTrigger>()))
+                .Callback<long, ScheduledTrigger>((id, trigger) =>
+                {
+                    trigger.Id = 3;
+                    trigger.JobId = 4;
+                });
+
+            // Act
+            _service.Add(1, triggerModel);
+
+            // Assert
+            triggerModel.Id.ShouldBe(3);
+            triggerModel.JobId.ShouldBe(4);
+        }
+
+        [TestMethod]
+        public void Add_ScheduledTrigger_ShouldCallMessengerHubPublish()
+        {
+            // Arrange
+            var model = new ScheduledTriggerModel { Id = 1, JobId = 2 };
+
+            // Act
+            _service.Add(1L, model);
+
+            // Assert
+            _messengerHubMock.Verify(hub => hub.PublishAsync(It.Is<TriggerAddedMessage>(message => message.TriggerId == 1 && message.JobId == 2)), Times.Once());
+        }
+
+        [TestMethod]
+        public void Add_InstantTrigger_ShouldSetModelKeys()
+        {
+            // Arrange
+            var triggerModel = new InstantTriggerModel { Id = 1, JobId = 2 };
+            _repositoryMock
+                .Setup(rep => rep.SaveAddTrigger(It.IsAny<long>(), It.IsAny<InstantTrigger>()))
+                .Callback<long, InstantTrigger>((id, trigger) =>
+                {
+                    trigger.Id = 3;
+                    trigger.JobId = 4;
+                });
+
+            // Act
+            _service.Add(1, triggerModel);
+
+            // Assert
+            triggerModel.Id.ShouldBe(3);
+            triggerModel.JobId.ShouldBe(4);
+        }
+
+        [TestMethod]
+        public void Add_InstantTrigger_ShouldCallMessengerHubPublish()
+        {
+            // Arrange
+            var model = new InstantTriggerModel { Id = 1, JobId = 2 };
+
+            // Act
+            _service.Add(1L, model);
+
+            // Assert
+            _messengerHubMock.Verify(hub => hub.PublishAsync(It.Is<TriggerAddedMessage>(message => message.TriggerId == 1 && message.JobId == 2)), Times.Once());
+        }
+
+        [TestMethod]
+        public void Disable_ShouldCallRepository()
+        {
+            // Arrange
+            long jobId = 1L, triggerId = 2L;
```

Comment:
Initialize every variable on its own line.
Consider enforcing this via StyleCop

Reply:
Interesting, I also was unable to find such a rule.